apis/kueue/v1beta1/resourceflavor_types.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+//+kubebuilder:object:root=true
+//+kubebuilder:resource:scope=Cluster,shortName={rf}
+//+kubebuilder:storageversion
+
+// ResourceFlavor is the Schema for the resourceflavors API.
+type ResourceFlavor struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
```

Comment:
Why no Spec and Status?

Reply:
So far we haven't found a case for Status, so we skipped it for simplicity. Can you think of something?

This is similar to how PriorityClasses are defined https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/#priorityclass

Reply:
I think that we will have a problem IF we find a use case for status (like resource depleted notification or so). Without Spec and Status we will have to do another api version, deprecation, migration etc just to have that bit of info exposed. Empty Status doesn't cost us much at this moment, but may be very handy in the future.

Reply:
I guess we don't need to add a status field, but we can add the spec. See update.